src/str.ts

```diff
@@ -425,6 +425,15 @@ export class Str {
     return this.value === this.upper().get()
   }
 
+  /**
+   * Determine whether the given string includes capital letters excluding the first character.
+   *
+   * @returns {Boolean}
+   */
+  hasUppercaseWithin (): boolean {
+    return /[A-Z]([A-Z0-9]*[a-z][a-z0-9]*[A-Z]|[a-z0-9]*[A-Z][A-Z0-9]*[a-z])[A-Za-z0-9]*/g.test(this.toString())
```

Comment:
The regex looks complex and I don’t want to maintain that. Can you think of another way for the check?

Reply:
Regex's are not my forte, the only optimisation I can find for this is by replacing the final `[A-Za-z0-9]` with `.` but I am not sure if this will be enough for you. Thoughts?

Reply:
Here’s what I understand this method should do: check if any other character than the first one is uppercase. That’s the case when the slice of the string starting at the second character is not the same as its lowercase version.

```suggestion
    const slice = new Str(this.value.slice(1))
    
    return slice.equals(
      slice.lower().get()
    )
```

Do you agree?